src/Widgets/SeekBar.vala

```diff
@@ -61,35 +61,35 @@ public class Music.SeekBar : Gtk.Grid {
         };
 
         scale = new Gtk.Scale.with_range (Gtk.Orientation.HORIZONTAL, 0, 1, 0.1) {
+            css_classes = {"accent"},
             draw_value = false,
             hexpand = true
         };
-        scale.get_style_context ().add_class (Granite.STYLE_CLASS_ACCENT);
 
-        scale.button_press_event.connect (() => {
-            scale.value_changed.connect (scale_value_changed);
+        // scale.button_press_event.connect (() => {
+        //     scale.value_changed.connect (scale_value_changed);
 
-            scale_pressed = true;
-            return Gdk.EVENT_PROPAGATE;
-        });
+        //     scale_pressed = true;
+        //     return Gdk.EVENT_PROPAGATE;
+        // });
 
-        scale.button_release_event.connect (() => {
-            scale.value_changed.disconnect (scale_value_changed);
-            PlaybackManager.get_default ().seek_to_progress (scale.get_value ());
-            scale_pressed = false;
-            return Gdk.EVENT_PROPAGATE;
-        });
+        // scale.button_release_event.connect (() => {
+        //     scale.value_changed.disconnect (scale_value_changed);
+        //     PlaybackManager.get_default ().seek_to_progress (scale.get_value ());
+        //     scale_pressed = false;
+        //     return Gdk.EVENT_PROPAGATE;
+        // });
 
         column_spacing = 6;
-        get_style_context ().add_class (Granite.STYLE_CLASS_SEEKBAR);
-        add (position_label);
-        add (scale);
-        add (duration_label);
+        css_classes = {"seekbar"};
+        attach (position_label, 0, 0);
+        attach (scale, 1, 0);
+        attach (duration_label, 2, 0);
```

Comment:
This is exactly what I meant with box:

```vala
append (position_label);
append (scale);
append (duration_label);
```

etc. And you have `prepend()` and `insert()` too. Grid is more flexible as it handles, well, grids, but is also way more complex to use.